### PR TITLE
Update API documentation referencing removed methods

### LINKED ISSUE #6641: Update documentation that refers to removed APIs
For example, `activateFetched` still has four references, which will make it challenging if developers need to replace it.

```
FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:/// Remote Config Server until they are copied to the Active Config by calling activateFetched. When
FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:/// they are copied to the Active Config by calling activateFetched. When you fetch values from the
FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:/// Fetches Remote Config data with a callback. Call activateFetched to make fetched data available
FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:/// Call activateFetched to make fetched data available to your app.
```

### PR DESCRIPTION
**Pull request by karenyz**
Fixes #6641

### CODE REVIEW

File: FirebaseRemoteConfig/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 - [changed] Functionally neutral change to stop using a deprecated method in the AB Testing API. (#6543)
 - [fixed] Updated `numberValue` to be nonnull to align with current behavior. (#6623)
 - [removed] Removed deprecated APIs `isDeveloperModeEnabled`, `initWithDeveloperModeEnabled:developerModeEnabled`, `activateWithCompletionHandler:completionHandler`, `activateFetched`, `configValueForKey:namespace`, `configValueForKey:namespace:source`, `allKeysFromSource:namespace`, `keysWithPrefix:namespace`, `setDefaults:namespace`, `setDefaultsFromPlistFileName:namespace`, `defaultValueForKey:namespace`. (#6637)
+- [changed] Updated public API documentation to no longer reference removed APIs. (#6667)
```
Comment (paulb777) [inline FirebaseRemoteConfig/CHANGELOG.md:6]: Prefer Issue # over PR # when available for Changelogs

**Review by paulb777 — COMMENTED** [review]
The changes look good, but should ideally be Swift-first for documenting other APIs.

I'll defer to @morganchen12 if that should be done now or in a later PR.

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h [hunk 2]
```diff
@@ -154,20 +154,21 @@ NS_SWIFT_NAME(RemoteConfig)
...
 
 /// Returns the FIRRemoteConfig instance for your (non-default) Firebase appID. Note that Firebase
 /// analytics does not work for non-default app instances. This singleton object contains the
 /// complete set of Remote Config parameter values available to the app, including the Active Config
 /// and Default Config. This object also caches values fetched from the Remote Config Server until
-/// they are copied to the Active Config by calling activateFetched. When you fetch values from the
-/// Remote Config Server using the default Firebase namespace service, you should use this class
-/// method to create a shared instance of the FIRRemoteConfig object to ensure that your app will
-/// function properly with the Remote Config Server and the Firebase service.
+/// they are copied to the Active Config by calling `activateWithCompletion`. When you fetch values
```
Comment (chliangGoogle) [inline FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h:168]: nit: `activateWithCompletion:` and the same for the followings.

**Review by chliangGoogle — CHANGES_REQUESTED** [review]

File: FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h (hunk 2 above)
Comment (karenyz) [reply FirebaseRemoteConfig/Sources/Public/FirebaseRemoteConfig/FIRRemoteConfig.h]: updated

**Comment by morganchen12** [comment]
Ideally we should use Swift selector syntax (so `activate(completion:)`), but we can do that in a different PR.

**Review by paulb777 — APPROVED** [review]

**Review by chliangGoogle — APPROVED** [review]

**Pull request merged**